metrics/bin/metrics.dart

```diff
@@ -0,0 +1,163 @@
+    _log.info('Skipping metrics upload');
+    return;
+  }
+
+  try {
+    await uploadMetrics(options, pubspec);
+  } catch (e, s) {
+    _log.warning('Failed to upload metrics', e, s);
+    // We don't set exitCode > 0 on runtime errors!
+    // This script is called during build (via gradle, podspec, etc.)
+    // and we don't want to be the cause of a broken build!
+  }
+  exit(0); // why is this needed?
+}
+
+Future<void> uploadMetrics(Options options, Pubspec pubspec) async {
+  final hostId = await machineId();
+
+  final metrics = await generateMetrics(
+    distinctId: hostId,
+    targetOsType: options.targetOsType,
+    targetOsVersion: options.targetOsVersion,
+    anonymizedMacAddress:
+        hostId, // cannot get this with dart, using hostId instead :-/ (similar to realm-js)
+    anonymizedBundleId: pubspec.name.strongHash(),
+  );
+
+  const encoder = JsonEncoder.withIndent('  ');
+  final payload = encoder.convert(metrics.toJson());
+  _log.info('Uploading metrics for ${pubspec.name}...\n$payload');
+  final base64Payload = base64Encode(utf8.encode(payload));
+
+  final client = HttpClient();
+  final request = await client.getUrl(
+    Uri.parse(
+      'https://webhooks.mongodb-realm.com'
+      '/api/client/v2.0/app/realmsdkmetrics-zmhtm/service/metric_webhook/incoming_webhook/metric'
+      '?data=$base64Payload}',
+    ),
+  );
+  await request.close();
+}
+
+// log to stdout
+final _log = Logger('metrics')
+  ..onRecord.listen((record) {
+    stdout.writeln('[${record.level.name}] ${record.message}');
+  });
+
+Future<Digest> machineId() async {
+  String? id;
+  try {
+    // No easy access to mac-address from dart, as used by other SDKs,
+    // but we can do better..
+    if (Platform.isLinux) {
+      // For linux use /etc/machine-id
+      // Can be changed by administrator but with unpredictable consequences!
+      // (see https://man7.org/linux/man-pages/man5/machine-id.5.html)
+      final process = await Process.start('cat', ['/etc/machine-id']);
+      id = await process.stdout.transform(utf8.decoder).join();
+    } else if (Platform.isMacOS) {
+      // For MacOS, use the IOPlatformUUID value from I/O Kit registry in
+      // IOPlatformExpertDevice class
+      final process = await Process.start('ioreg', [
+        '-rd1',
+        '-c',
+        'IOPlatformExpertDevice',
+      ]);
+      id = await process.stdout.transform(utf8.decoder).join();
+      final r = RegExp('"IOPlatformUUID" = "([^"]*)"', dotAll: true);
+      id = r.firstMatch(id)?.group(1); // extract IOPlatformUUID
+    } else if (Platform.isWindows) {
+      // For Windows, use the key MachineGuid in registry:
+      // HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Cryptography
+      // Can be changed by administrator but with unpredictable consequences!
+      //
+      // It is generated during OS installation and won't change unless you make
+      // another OS update or re-install. Depending on the OS version it may
```

Comment:
could this change with an OS update. Its quite common in Windows nowadays to have short release cycles.
This is how Realm JS is doing it using "Windows management instrumentation" 
https://github.com/realm/realm-js/blob/fea4a664fa372005f8351c1f628d00eff614f8e8/react-native/android/analytics.gradle#L183

Reply:
I think it would be one of the harder bits to change fro MS, and not all vendors supply a UUID via wmic (https://social.technet.microsoft.com/Forums/scriptcenter/en-US/2bd70996-8dbf-4a96-85b0-341b9cf2ad74/wmic-uuid-returns-ffffffffffffffffffffffffffffffff?forum=ITCG), so I think this is a better id.

Reply:
About the Winodws update if this Id changes than the old builder will be counted as a new one. We should check how likely is this to change on Windows update.

The wmi comment is just a reference to compare how Realm JS is doing this. I agree and don't think it's better

Reply:
Will use as is for now. May revisit later